06_neighborhoods/environment/air_toxicity_metric.qmd

```diff
@@ -0,0 +1,855 @@
+```{r}
+final_data_county_sub  <- final_data_county  %>% 
+  mutate(index_air_hazard = round(index_air_hazard, digits = 0))
+```
+
+Create all version of data.
+```{r}
+final_data_county_all <- final_data_county_sub %>%
+  filter(subgroup == "All")
+```
+
+## Crosswalk place
+
+Crosswalk the individual years of data onto the UMF geographic tract crosswalk for places file taken from the geographic-crosswalks folder. This will create a version of the data where all tracts that fall within places used in the main UMF data set are present in the final data (regardless of missing EPA data). Note that the EPA did not change which counties are included between 2014, 2018 and 2019.
+```{r}
+
+crosswalk_place <- read_csv(here::here("geographic-crosswalks", "data", "geocorr2018_tract_to_place.csv")) %>% 
+  mutate(county = str_sub(county, start = -3L, end = -1L),
+         state = str_pad(state, 2, side = "left", pad = "0"),
+         tract = tract * 100,
+         tract = str_pad(tract, 6, side = "left", pad = "0"),
+         tract = paste0(state, county, tract),
+         year1 = "2019",
+         year2 = "2018",
+         year3 = "2014") %>% 
+  select(place = placefp, tract, afact,  year1:year3) %>% 
+  pivot_longer(!c(place, tract, afact), values_to = "year", names_to = "value") %>% 
+  select(-value)
+
+```
+
+Pad the place variable with a leading zero.
+```{r}
+crosswalk_place <- crosswalk_place %>% 
+  mutate(place = str_pad(place, 5, side = "left", pad = "0")) %>% 
+  select(year, place, tract, afact)
+```
+
+Merge the environmental index ACS data onto the city crosswalk file.
+```{r}
+tract_place_hazard <- crosswalk_place %>% 
+  left_join(haz_index_all, by = c("year", "tract")) %>%
+  mutate(place = as.character(place),
+         state = str_sub(tract, 1, 2)) %>% 
+  mutate(place_id = str_c(state, place))
+
+anti_join(crosswalk_place, haz_index_all, by = c("year", "tract")) %>% 
+  mutate(state = str_sub(tract, start = 1, end = 2)) %>% 
+  group_by(state, year) %>%
+  count() %>% 
+  arrange(desc(n))
+```
+
+Merge on ACS information.
+```{r}
+haz_index_acs <- tract_place_hazard %>% 
+  left_join(race_pov, by=c("tract" = "geoid", "year")) %>% 
+  mutate(na_pop= if_else(is.na(index_air_hazard), total_pop, 0))
+```
+
+## Create place level index
+
+Import places/pop file and prep to limit to population cutoff places.
+```{r}
+state_place_pop <- read_csv(here::here("geographic-crosswalks", "data", "place-populations.csv")) %>% 
+  mutate(place_id = paste0(state, place))
+```
+
+Create "All" observations calculate place population for tracts and index_air_hazard weighting by amount of tract in place and tract total population
+
+```{r}
+all_enviro_place <- haz_index_acs %>%
+  mutate(tract_pop = (total_pop*afact)) %>% #to account for the fact that only part of the tract population is in the place
+  group_by(year, place_id, state, place) %>%
+  summarise(index_air_hazard = weighted.mean(index_air_hazard, tract_pop, na.rm = TRUE),
```

Comment:
Looking back at this I think this was incorrect, I should be incorporating afact into the place_pop calculation. Will make the change.

Reply:
If I am looking at this correctly, you have the variable `tract_pop` which is the adjusted `total_pop` but then in the summarize statement you have the variables multiplied by `tract_pop` - is that double adjusting? Or is this what you intended?

Reply:
I think this is correct but if you could confirm my logic aligns with the code that would be great.
The idea is that tract_pop is the weighted population of the tract (created by multiplying the share of the tract pop in a given place by the total tract pop). The weighted mean calculation uses tract population to weight the metric up to the place mean, we want to use the weighted tract population in this case because not all of these tracts fit in the place.

Reply:
I think this logic is correct but I'm still not sure about this line of code specifically:

`place_pop = sum(tract_pop * afact))`

Say you have a tract of population 1,000 that is 50% in one place and 50% in another. If you create tract_pop the value of that will be 500. If you then create place pop by taking tract_pop times 50% you would only be summing 250 rather than 500 and thus in this case undercounting your place population by half. 

I may be wrong and that's not what this line of code is doing. If that's the case I think we can call this resolved.

Reply:
I think it's possible you are referencing an older version but the way this is written the tract_pop varaible used in the place_pop sum is not multiplied a second time by afact. I agree with your logic that we would be underestimating population if that was the case - but the way this code is written tract_pop is created by taking total_pop and multiplying by afact. Then this variable is summed to the place level (no additional use of the AFACT varaible). So to use your example we get the entire 500 as opposed to 250.

Reply:
I am on branch iss211 and think I have the most recent edits pulled. In R studio what I see as line 791 is what I think is still double multiplying. Conversely line 835 is one where it looks like you made the proper edit.

Reply:
@rpitingolo I was looking at the version this comment was attached to which was old, you are exactly right thank you for flagging. I updated to correct that.